test/sqllogictest/transform/join_fusion.slt

```diff
@@ -116,8 +116,8 @@ Explained Query:
     Filter (#5 != #11) AND (#14 >= #10) // { arity: 15 }
       Join on=(#0 = #8 AND date_to_timestamp(#11) = (#5 - 9 months)) type=differential // { arity: 15 }
         implementation
-          %2:customer[×] » %0:lineitem[×]A » %1:orders[#0, date_to_timestamp(#3)]KKA
-        ArrangeBy keys=[[]] // { arity: 8 }
+          %0:lineitem[#0, (#5 - 9 months)]KKA » %1:orders[#0, date_to_timestamp(#3)]KKA » %2:customer[×]A
+        ArrangeBy keys=[[#0, (#5 - 9 months)]] // { arity: 8 }
```

Comment:
I'm a little confused what happened here. Was the previous plan broken, choosing an arrangement that wouldn't actually work correctly for the join?

Reply:
Ah, or is this your second point about the fixed bug wrt the characteristics being wrong (though this is the arrangement itself, not the characteristics)?

Reply:
I think the confusing thing here is that the key that we print for the starting input has a different meaning from the keys that we print for the other inputs:
- The key that we print for the starting input is the key that is used for the join with the next input. (`stream_key` in the terminology of `LinearJoinPlan::create_from`)
- The key that we print for the other inputs is the key that is used for the join with the previous input. (`lookup_key`)

So, despite appearances, `lineitem`'s key when joining it with `orders` is actually the same in both the previous plan and the new plan.

Maybe it was a bad idea to print the `start_key` at all? I'm not sure. I can see 3 options now:
1. not print the `start_key`, or
2. print both `stream_key` and `lookup_key` at every stage, or
3. leave it as it is.

I don't like 1., because then EXPLAIN is keeping a field of our `JoinImplementation` struct a secret.

But 2. is also problematic, because `stream_key` is not actually recorded in the MIR `JoinImplementation` struct. (`LinearJoinPlan::create_from` figures it out from looking at the `equivalences` again.)

And 3. would continue to be a source of confusion for others.

I don't like any of these options... What do you think?

Reply:
Thinking about this again, I'd say let's merge it as it is, and later tweak the EXPLAIN output a bit. (Maybe we can put `< ... >` in front of the starting input's keys to differentiate it from the `[ ... ]` of later inputs.)